repl.js

```diff
@@ -293,13 +300,21 @@ class CombeeQueue {
         end: i + BATCH_SIZE - 1,
       });
 
-      const matched = jobs.filter(filterFn);
-      if (matched && matched.length) {
-        matches = matches.concat(matched);
+      for (const job of jobs) {
+        yield job;
       }
     }
+  }
 
-    return matches;
+  /**
+   * @param {JobType} jobType
+   * @param {function(BeeQueue.Job)} fn The function to execute on each job of the given type
+   * @returns {Promise<void>}
+   */
+  async forEach(jobType, fn) {
```

Comment:
Thoughts on adding an optional filter function here so that users could down filter the jobs that fn is run on? They could implement it in fn, and it’d be nice if they didn’t need to logic through that if they have a quick filter they’d like to apply.

Reply:
Good call, thoughts on f27f60af7221d149afdb1b622528a3bbd86946a4? This allows for usage like:

```
>combee.someQueue.iterate('waiting').filter({ 'data.user': 'foo' }).forEach((job) => job.remove()).then(() => console.log('done'))
```

Reply:
I like it, this would be a great example to call out in the docs in a separate PR.

Reply:
https://github.com/ttacon/combee/pull/16